app/templates/contacts/new.hbs

```diff
@@ -0,0 +1,131 @@
+{{page-title "Nieuwe verzendlijst"}}
+<Auk::Navbar @skin="gray-100" @size="large">
+  <Auk::Toolbar>
+    <Auk::Toolbar::Group  @position="left">
+      <Auk::Toolbar::Item>
+        <Auk::Button
+          @layout="icon-only"
+          @skin="borderless"
+          @icon="chevron-left"
+          @size="large"
+          {{on "click" this.navigateBack}}
+        />
+      </Auk::Toolbar::Item>
+      <Auk::Toolbar::Item>
+        <div class="auk-overline auk-u-block auk-u-muted">
+          Verzendlijsten
+        </div>
+        <h2 class="auk-toolbar__title">
+          Nieuwe verzendlijst
+        </h2>
+      </Auk::Toolbar::Item>
+    </Auk::Toolbar::Group>
+  </Auk::Toolbar>
+</Auk::Navbar>
+
+<div class="auk-u-m-8">
+  <Auk::ProgressBar @progress={{this.progress}} @disablePercentage={{true}}/>
+  <div class="auk-o-flex auk-o-flex--align-baseline auk-o-flex--justify-between">
+    <h1 class="auk-h1">{{this.title}}</h1>
+    {{#if (and (eq this.step 2) this.inputType)}}
+      <Auk::Button
+        @skin="secondary"
+        @icon="add"
+        {{on "click" this.openContactItemModal}}>
+          Contactpersoon toevoegen
+      </Auk::Button>
+    {{/if}}
+  </div>
+  <div>
+    {{#if (eq this.step 0)}}
+      <div class="auk-o-grid-col-6 auk-form-group">
+        <Auk::Label for="name">
+          Titel verzendlijst
+        </Auk::Label>
+        <Auk::Input
+          type="text"
+          id="name"
+          @block={{true}}
+          value={{this.model.name}}
+          {{on "input" (fn this.setInputValue this.model "name")}}
+        />
+      </div>
+    {{/if}}
+    {{#if (eq this.step 1)}}
+      <div class="auk-o-grid-col-6 auk-form-group">
+        <Auk::Label for="name">
+          Kies een input type
+        </Auk::Label>
+        <Auk::RadioList as |list|>
+          <list.Radio
+            @label="CSV of tabel tekst bestand"
+            @sublabel="Importeer ontvangers van een .csv of .txt bestand"
+            @checked={{(eq this.inputType "csv")}}
+            value="csv"
+            {{on "change" this.setInputType}}/>
+          <list.Radio
+            @label="Manuele input"
+            @sublabel="Geef de contact informatie per ontvanger manueel in"
+            @checked={{(eq this.inputType "manual")}}
+            value="manual"
+            {{on "change" this.setInputType}}/>
+        </Auk::RadioList>
+      </div>
+    {{/if}}
+    {{#if (eq this.step 2)}}
+      {{#if (eq this.inputType "manual")}}
+        <ContactItem::Table
+          @onOpenContactItem={{this.openContactItem}}
+          @model={{this.contacts}}
+          @noDataMessage="Er zijn nog geen contactpersonen."
+        />
+      {{/if}}
+      {{#if showContactItemModal}}
+        <ContactItem::Modal
+          @isNewContact={{this.isNewContact}}
+          @contact={{this.selectedContact}}
+          @onCancel={{this.closeContactItemModalAndReset}}
```

Comment:
De rollback van de wijzigingen zou ik in de `modal.js` component uitvoeren, aangezien de wijzigingen daar ook toegepast worden. In `new.js` controller zou ik enkel nog het sluiten van de modal doen.

Het rollbacken is hier eenvoudiger dan bij persberichten/bronnen. Wanneer de gebruiker 'Annuleren' klikt, mogen wijzigingen sowieso ongedaan gemaakt worden. Er moet geen previous en huidige state vergeleken worden en er kunnen ook geen relaties wijzigen, dus de hele snapshot constructie is niet nodig. Je kan gewoon `rollbackAttributes` uitvoeren op alle modellen (het contact-item, de phone en het emailadres). Daarmee zouden alle wijzigingen ongedaan gemaakt moeten zijn.